cpp/oneapi/dal/table/detail/csr_block.hpp

```diff
@@ -0,0 +1,42 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "oneapi/dal/table/common.hpp"
+
+namespace oneapi::dal::detail {
+namespace v1 {
+
+/// Type of csr indexing, currently it's implemented only `one_based` indexing.
+enum class csr_indexing { zero_based, one_based };
```

Comment:
Why this is placed in this file?

Reply:
We may discuss offline, the idea is that enum and struct are used in csr.hpp and inner headers csr_access_iface.hpp (and in other places)

Reply:
If there is any objection, it might be applied later